app/jobs/scheduled/badge_grant.rb

```diff
@@ -22,6 +22,7 @@ def execute(args)
       end
 
       BadgeGranter.revoke_ungranted_titles!
+      UserBadge.ensure_consistency! # Badge granter sometimes uses raw SQL, so hooks do not run. Clean up data
```

Comment:
This worries me a bit, what is the cost of the new big query on one of our larger sites?